### PR TITLE
feat: tracing for console wallet

### PR DESCRIPTION
**Pull request by StriderDM**
Description
---
This PR adds instrumentation to methods within the tari_console_wallet for tracing output in Jaeger. 

Also includes some clippy fixes with the latest update to the Rust toolchain

Motivation and Context
---
This will give an indication of how often which method is called and from which component.

How Has This Been Tested?
---
Manually

### CODE REVIEW

File: base_layer/wallet/src/connectivity_service/handle.rs [hunk 1]
```diff
@@ -58,6 +63,7 @@ impl WalletConnectivityHandle {
 
 #[async_trait::async_trait]
 impl WalletConnectivityInterface for WalletConnectivityHandle {
+    #[instrument(name = "wallet_connectivity_interface::set_base_node", skip(self, base_node_peer))]
     fn set_base_node(&mut self, base_node_peer: Peer) {
```
Comment (stringhandler) [inline base_layer/wallet/src/connectivity_service/handle.rs]: This method will clearly return instantly. There's no need to instrument it.

**Review by stringhandler — COMMENTED** [review]
Feels like you have just applied the #[instrument] tag to every method without thinking about whether the method returns instantly or not. Have you had a look at the output? How many of these methods are on a critical path. I would suggest looking at the jaeger output and seeing which methods are useful.

I am worried that if we merge this PR we'll be removing these attributes forever.

File: base_layer/wallet/src/base_node_service/service.rs [hunk 2]
```diff
@@ -86,11 +87,13 @@ where T: WalletBackend + 'static
     }
 
     /// Returns the last known state of the connected base node.
+    #[instrument(name = "base_node_service::get_state", skip(self))]
     pub async fn get_state(&self) -> BaseNodeState {
         self.state.read().await.clone()
     }
 
     /// Starts the service.
+    #[instrument(name = "base_node_service::start", skip(self))]
```
Comment (philipr-za) [inline base_layer/wallet/src/base_node_service/service.rs]: Cut this one

File: base_layer/wallet/src/base_node_service/handle.rs [hunk 3]
```diff
@@ -65,17 +66,20 @@ impl BaseNodeServiceHandle {
         }
     }
 
+    #[instrument(name = "base_node_service_handle::get_event_stream", skip(self))]
```
Comment (philipr-za) [inline base_layer/wallet/src/base_node_service/handle.rs]: cut

File: base_layer/wallet/src/base_node_service/monitor.rs [hunk 4]
```diff
@@ -194,6 +196,7 @@ where
         Ok(())
     }
 
+    #[instrument(name = "base_node_monitor::map_state", skip(self, transform))]
```
Comment (philipr-za) [inline base_layer/wallet/src/base_node_service/monitor.rs]: cut, an in-memory operation

File: base_layer/wallet/src/base_node_service/monitor.rs [hunk 5]
```diff
@@ -205,6 +208,7 @@ where
         self.publish_event(BaseNodeEvent::BaseNodeStateChanged(new_state));
     }
 
+    #[instrument(name = "base_node_monitor::publish_event", skip(self, event))]
```
Comment (philipr-za) [inline base_layer/wallet/src/base_node_service/monitor.rs]: cut

File: base_layer/wallet/src/base_node_service/service.rs [hunk 6]
```diff
@@ -86,11 +87,13 @@ where T: WalletBackend + 'static
     }
 
     /// Returns the last known state of the connected base node.
+    #[instrument(name = "base_node_service::get_state", skip(self))]
```
Comment (philipr-za) [inline base_layer/wallet/src/base_node_service/service.rs]: Cut, in memory operation

File: base_layer/wallet/src/base_node_service/service.rs [hunk 7]
```diff
@@ -121,6 +124,7 @@ where T: WalletBackend + 'static
         Ok(())
     }
 
+    #[instrument(name = "base_node_service::spawn_monitor", skip(self))]
```
Comment (philipr-za) [inline base_layer/wallet/src/base_node_service/service.rs]: cut, this just spawn. No major work done.

File: base_layer/wallet/src/connectivity_service/handle.rs [hunk 8]
```diff
@@ -67,6 +69,7 @@ impl WalletConnectivityInterface for WalletConnectivityHandle {
         self.base_node_watch.send(Some(base_node_peer));
     }
 
+    #[instrument(name = "wallet_connectivity_interface::get_current_base_node_watcher", skip(self))]
```
Comment (philipr-za) [inline base_layer/wallet/src/connectivity_service/handle.rs]: cut

File: base_layer/wallet/src/connectivity_service/handle.rs [hunk 9]
```diff
@@ -110,26 +118,35 @@ impl WalletConnectivityInterface for WalletConnectivityHandle {
         reply_rx.await.ok()
     }
 
+    #[instrument(name = "wallet_connectivity_interface::get_connectivity_status", skip(self))]
```
Comment (philipr-za) [inline base_layer/wallet/src/connectivity_service/handle.rs]: cut

File: base_layer/wallet/src/connectivity_service/handle.rs [hunk 10]
```diff
@@ -110,26 +118,35 @@ impl WalletConnectivityInterface for WalletConnectivityHandle {
...
     fn get_connectivity_status(&mut self) -> OnlineStatus {
         *self.online_status_rx.borrow()
     }
 
+    #[instrument(name = "wallet_connectivity_interface::get_connectivity_watch", skip(self))]
```
Comment (philipr-za) [inline base_layer/wallet/src/connectivity_service/handle.rs]: cut

**Review by philipr-za — COMMENTED** [review]
Ok, I still think there are many methods instrumented here that do not need to be. I started marking a few and mentioning why they shouldn't be instrumented but I am not going to do them all. Please apply those criterion to the rest.

Any method that is just:
- spawning a task (returns immediately),
- returning a channel,
- returning a piece of data from memory,
should not be instrumented.

What should be instrumented:
- Any method that does any appreciable work
- Any method that makes use of a scarce resource like the database or network

File: base_layer/wallet/src/output_manager_service/master_key_manager.rs [hunk 11]
```diff
@@ -142,6 +143,7 @@ where TBackend: OutputManagerBackend + 'static
 
     /// Return the next pair of (spending_key, script_private_key) from the key managers. These will always be generated
     /// in tandem and at corresponding increments
+    #[instrument(name = "key_manager::get_next_spend_and_script_key", skip(self))]
```
Comment (stringhandler) [inline base_layer/wallet/src/output_manager_service/master_key_manager.rs]: I would not add any tracing to this file at the risk of leaking information

Comment (StriderDM) [reply base_layer/wallet/src/output_manager_service/master_key_manager.rs]: Have removed them, thanks.

**Review by philipr-za — DISMISSED** [review]
LGTM

**Review by philipr-za — APPROVED** [review]

**Pull request merged**